app/controllers/course/survey/surveys_controller.rb

```diff
@@ -84,7 +84,8 @@ def survey_params
       :title, :description, :base_exp, :time_bonus_exp, :start_at, :bonus_end_at, :end_at,
       :published, :allow_response_after_end, :allow_modify_after_submit
     ]
-    fields << :anonymous if !@survey.anonymous || !@survey.has_student_response?
+    fields << :anonymous if action_name == 'create' || !@survey.anonymous ||
```

Comment:
@kxmbrian that depends, I'm scared we change stuff on production at `rails c` that we are not aware about. If there are other services that modify the survey attributes, it would not be caught by whitelisting params. 

How critical is it if this `anonymous` field is changed? If not so critical, I think we can leave it as is. If not, I would suggest to add the validation.